circuits/cpp/src/aztec3/circuits/kernel/public/common.cpp

```diff
@@ -29,6 +29,10 @@ void common_initialise_end_values(PublicKernelInputs<NT> const& public_kernel_in
 
     end.public_data_update_requests = start.public_data_update_requests;
     end.public_data_reads = start.public_data_reads;
+
+    // Public kernel does not modify encrypted logs values --> we just copy them to output
+    end.encrypted_logs_hash = start.encrypted_logs_hash;
```

Comment:
I've probably misunderstood something but why do we need these in public kernel?

Reply:
It's because public kernel can be chained with private kernel. So if there was public kernel after private the encrypted logs hash would not get passed through to the next kernel or base rollup circuit.